### DISCUSSION TITLE
Masking connection names in task logs

### DISCUSSION THREAD

**Discussion by arjunanan6** [apache/airflow#34542] [Q&A] [answered]
I have a requirement where I need to mask the connection ID from appearing in the task logs. 

Example
[2023-09-22, 05:01:22 CEST] {base.py:73} INFO - Using connection ID 'some_conn' for task execution.

Is there a way to hide this `some_conn` from appearing in the task logs?

**Comment by Taragolis** [comment, marked as answer]
> Is there a way to hide this some_conn from appearing in the task logs?

I don't think it is possible to do that in the global level. Better you could do it manually add it to the masker.

```python
@task
def my_func(conn_id):
    from airflow.utils.log.secrets_masker import mask_secret

    mask_secret(conn_id)
```

It would still not work well with templated fields, and in additional for regular operators you need to overwrite all of them

**Reply by arjunanan6** [reply]
Thanks @Taragolis, I think that works. Templated fields are okay, the problem is that the logs from the cluster go into our splunk log observer and there, the conn ID is in plain sight - which security does not like. 

As long as I can prevent it from appearing in the task logs, this works perfectly.

**Reply by Taragolis** [reply]
Just for a record `conn_id` it is just a primary key (string) in connections table or in Secrets Backend, by it's nature it is not contain anything secret. However it might contain some PII data e.g. `john_doe_connection` or `awesome_customer_connection` so maybe if it your case it's better to switch to some anonymous connection_ids? like `1sda2sd345` or `5d09b739-0552-4622-9f22-91b6efab1ed4`
